usaspending_api/agency/tests/integration/test_sub_agency.py

```diff
@@ -0,0 +1,256 @@
+
+@pytest.mark.django_db
+def test_all_categories(client, monkeypatch, transaction_search_1, elasticsearch_transaction_index):
+    setup_elasticsearch_test(monkeypatch, elasticsearch_transaction_index)
+    resp = client.get(url.format(toptier_code="001", filter="?fiscal_year=2021"))
+
+    expected_results = [
+        {
+            "name": "Sub-Agency 1",
+            "abbreviation": "A1",
+            "total_obligations": 515,
+            "transaction_count": 5,
+            "new_award_count": 4,
+            "children": [
+                {"name": "Office 2", "total_obligations": 312.0, "transaction_count": 3, "new_award_count": 3},
+                {"name": "Office 1", "total_obligations": 203.0, "transaction_count": 2, "new_award_count": 1},
+            ],
+        }
+    ]
+    assert resp.status_code == status.HTTP_200_OK
+    assert resp.json()["results"] == expected_results
+
+
+@pytest.mark.django_db
+def test_alternate_year(client, monkeypatch, transaction_search_1, elasticsearch_transaction_index):
+    setup_elasticsearch_test(monkeypatch, elasticsearch_transaction_index)
+    resp = client.get(url.format(toptier_code="001", filter="?fiscal_year=2020"))
+    assert resp.status_code == status.HTTP_200_OK
+
+    expected_results = [
+        {
+            "name": "Sub-Agency 1",
+            "abbreviation": "A1",
+            "total_obligations": 300.0,
+            "transaction_count": 1,
+            "new_award_count": 1,
+            "children": [
+                {"name": "Office 1", "total_obligations": 300.0, "transaction_count": 1, "new_award_count": 1}
+            ],
+        }
+    ]
+    assert resp.status_code == status.HTTP_200_OK
+    assert resp.json()["results"] == expected_results
+
+
+@pytest.mark.django_db
+def test_alternate_agency(client, monkeypatch, transaction_search_1, elasticsearch_transaction_index):
+    setup_elasticsearch_test(monkeypatch, elasticsearch_transaction_index)
+    resp = client.get(url.format(toptier_code="002", filter="?fiscal_year=2021"))
+    assert resp.status_code == status.HTTP_200_OK
+
+    expected_results = [
+        {
+            "name": "Sub-Agency 2",
+            "abbreviation": "A2",
+            "total_obligations": 400.0,
+            "transaction_count": 1,
+            "new_award_count": 0,
+            "children": [
+                {"name": "Office 2", "total_obligations": 400.0, "transaction_count": 1, "new_award_count": 0}
+            ],
+        }
+    ]
+    assert resp.status_code == status.HTTP_200_OK
+    assert resp.json()["results"] == expected_results
+
+
+@pytest.mark.django_db
+def test_invalid_agency(client, monkeypatch, transaction_search_1, elasticsearch_account_index):
+    resp = client.get(url.format(toptier_code="XXX", filter="?fiscal_year=2021"))
+    assert resp.status_code == status.HTTP_404_NOT_FOUND
+
+    resp = client.get(url.format(toptier_code="999", filter="?fiscal_year=2021"))
+    assert resp.status_code == status.HTTP_404_NOT_FOUND
```

Comment:
Can you throw in tests for `award_type_codes` and `agency_type` as well? I don't think `award_type_codes` is currently working (I think due to TinyShield and another comment) and tests should pick that up if it happens.

Reply:
added, also the type codes have been fixed